### PR TITLE
Show deepest common field path in recorded manage events

### PR DESCRIPTION
**Pull request by bergerx**
## Summary
- Recorded manage events previously only showed top-level fields (`spec`, `metadata`, `status`) per managedFields entry, which was too coarse to tell what actually changed.
- Now each entry reports the deepest common path shared by the touched fields, e.g. `spec.template` or `metadata.annotations`, collapsing back to a shared ancestor like `status` when sibling fields (`conditions`, `phase`) were touched together.
- `metadata.labels`/`metadata.annotations` are treated as terminal so individual label/annotation key names aren't appended to the path.

## Test plan
- [x] `go test ./...` passes (436 tests)
- [x] Added `TestFieldsV1Paths` covering single-nested field, status fork, mixed multi-branch, and labels/annotations edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/plugin/template_functions_static_test.go [hunk 1]
```diff
@@ -167,6 +167,66 @@ func TestSortMapListByKeysValueIsStableOnTies(t *testing.T) {
...
+				"f:metadata": map[string]interface{}{"f:labels": map[string]interface{}{".": struct{}{}}},
+				"f:spec":     map[string]interface{}{"f:template": map[string]interface{}{".": struct{}{}}},
+				"f:status":   map[string]interface{}{"f:conditions": map[string]interface{}{}},
+			},
+			want: []string{"metadata.labels", "spec.template", "status.conditions"},
+		},
+		{
+			name:     "single owned label stops at metadata.labels",
+			fieldsV1: map[string]interface{}{"f:metadata": map[string]interface{}{"f:labels": map[string]interface{}{"f:app": map[string]interface{}{}}}},
+			want:     []string{"metadata.labels"},
```
Comment (bergerx) [inline pkg/plugin/template_functions_static_test.go]: Shoulnt this be "metadata.labels.app"?

File: pkg/plugin/template_functions_static_test.go [hunk 2]
```diff
@@ -167,6 +167,66 @@ func TestSortMapListByKeysValueIsStableOnTies(t *testing.T) {
...
+		},
+		{
+			name: "single owned annotation stops at metadata.annotations",
+			fieldsV1: map[string]interface{}{"f:metadata": map[string]interface{}{"f:annotations": map[string]interface{}{
+				"f:deployment.kubernetes.io/revision": map[string]interface{}{},
+			}}},
+			want: []string{"metadata.annotations"},
```
Comment (bergerx) [inline pkg/plugin/template_functions_static_test.go]: Same. Shouldn't this be `metadata.annotations."deployment.kubernetes.io/revision"`?

File: pkg/plugin/template_functions_static_test.go (hunk 1 above)
Comment (bergerx) [reply pkg/plugin/template_functions_static_test.go]: Good catch — fixed in 1712c46: it now descends to `metadata.labels.app` for a single owned label, and only stops at `metadata.labels` once multiple sibling label keys are touched.

File: pkg/plugin/template_functions_static_test.go (hunk 2 above)
Comment (bergerx) [reply pkg/plugin/template_functions_static_test.go]: Fixed in 1712c46 — descends to `metadata.annotations."deployment.kubernetes.io/revision"`, with the key quoted since it contains dots that would otherwise look like further nesting.
